presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/MergeFilters.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.iterative.rule;
+
+import com.facebook.presto.sql.planner.PlanNodeIdAllocator;
+import com.facebook.presto.sql.planner.SymbolAllocator;
+import com.facebook.presto.sql.planner.iterative.Lookup;
+import com.facebook.presto.sql.planner.iterative.Rule;
+import com.facebook.presto.sql.planner.plan.FilterNode;
+import com.facebook.presto.sql.planner.plan.PlanNode;
+
+import java.util.Optional;
+
+import static com.facebook.presto.sql.ExpressionUtils.combineConjuncts;
+
+public class MergeFilters
+    implements Rule
+{
+    @Override
+    public Optional<PlanNode> apply(PlanNode node, Lookup lookup, PlanNodeIdAllocator idAllocator, SymbolAllocator symbolAllocator)
+    {
+        if (!(node instanceof FilterNode)) {
+            return Optional.empty();
+        }
+
+        FilterNode parent = (FilterNode) node;
+
+        PlanNode source = lookup.resolve(parent.getSource());
+        if (!(source instanceof FilterNode)) {
+            return Optional.empty();
+        }
+
+        FilterNode child = (FilterNode) source;
+
+        return Optional.of(
+                new FilterNode(
+                        parent.getId(),
+                        child.getSource(),
+                        combineConjuncts(parent.getPredicate(), child.getPredicate())));
```

Comment:
Swap the order of the conjuncts. Otherwise, it might lead to some filters that can throw exceptions on input to be applied before filters that would exclude those inputs (arguably, this is a problem that we have to solve more comprehensively, but we don't want this to make it worse).